### PR TITLE
Enable opentelemetry locally

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Support open-telemetry for our local development.

You can use `make -C local build start-all` then you can start building jobs and go to http://localhost:5601/app/apm/services/jenkins/transactions to see those transactions

## Why is it important?

Test https://github.com/cyrille-leclerc/opentelemetry-plugin

## Tasks

- [x] Configure docker for open-telemetry since It works for https://github.com/cyrille-leclerc/opentelemetry-plugin/blob/master/opentelemetry-collector/run-opentelemetry-collector.sh but it does not work for the container. Somehow the `elastic` exporter is not recognised

## Tests

Blue Ocean running locally

image

Caused the below APM transactions (with INFO log level, aka less number of echo steps)

transactions_without_debug_echo

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

File: local/configs/jenkins.yaml [hunk 1]
```diff
@@ -102,6 +102,10 @@ unclassified:
   gitscm:
     globalConfigName: username
     globalConfigEmail: username@example.com
+
+  openTelemetry:
+    endpoint: "localhost:4317"
+    useTls: false
```
Comment (v1v) [inline local/configs/jenkins.yaml:108]: quick and dirty and might require the plugin to be installed beforehand

File: local/configs/plugins.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 disable-github-multibranch-status
 filesystem_scm
 job-dsl
-ssh-agent
\ No newline at end of file
+opentelemetry::https://storage.googleapis.com/apm-ci-temp/plugins/opentelemetry.hpi
```
Comment (v1v) [inline local/configs/plugins.txt:4]: I uploaded the plugin after building it locally

**Comment by cyrille-leclerc** [comment]
Interesting, should we ignore some steps like the `echo` step?

**Comment by v1v** [comment]
> Interesting, should we ignore some steps like the echo step?

@cyrille-leclerc , this particular project got a DEBUB log level, so that's the reason we see all those echo steps. Probably `echo` is not something to worry about unless there is a bug with some interpolation then I guess it might be useful to be trapped, what do you think?

**Comment by cyrille-leclerc** [comment]
> > Interesting, should we ignore some steps like the echo step?
> 
> @cyrille-leclerc , this particular project got a DEBUB log level, so that's the reason we see all those echo steps. Probably `echo` is not something to worry about unless there is a bug with some interpolation then I guess it might be useful to be trapped, what do you think?

Sounds good to me, just an idea to keep in mind then.

**Review by cachedout — APPROVED** [review]

**Review by mdelapenya — APPROVED** [review]

**Comment by cyrille-leclerc** [comment]
FYI the plugin won't create spans for some steps that are not very interesting to monitor:"dir", "echo", "isUnix", "pwd" https://github.com/cyrille-leclerc/opentelemetry-plugin/commit/191cf48f7b3633b156a4518eef03ffe25fe93ad1

**Pull request merged**